pre-receive-hooks/block_unknown_pushers.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+#
+# Pre-receive hook that will block any pushes / repository modifications
+# not performed by a user in the list (foo, bar, foobar)
+#
+# More details on pre-receive hooks and how to apply them can be found on
+# https://help.github.com/enterprise/admin/guides/developer-workflow/managing-pre-receive-hooks-on-the-github-enterprise-appliance/
+#
+
+case $GITHUB_USER_LOGIN in
+  foo|bar|foobar) echo "User $GITHUB_USER_LOGIN is allowed to push";;
```

Comment:
Yes. I just tested and without `shopt -s extglob` and using the `+(foo|bar|foobar)` or `@(foo|bar|foobar)` syntax for detailing the case match it won't work. I think the description at the top is enough then.